chain-signatures/node/src/mesh/connection.rs

```diff
@@ -83,7 +101,12 @@ impl NodeConnection {
                 Ok(state) => {
                     let new_status = match state {
                         StateView::Running { .. } | StateView::Resharing { .. } => {
-                            NodeStatus::Active
+                            // The peer is running. But before we can reliably
+                            // use the connected node in protocols we initiate,
+                            // we need to ensure the peer has the up-to-date
+                            // data about out owned IDs.
+                            NodeStatus::Syncing
+                            // TODO: (Find out in review) Is my own ID also part of this list?
```

Comment:
I saw in other places that participants can include the own ID. Do we need to worry about this here? We wouldn't want to trigger a sync with ourselves... @ChaoticTempest

Reply:
yep, the mesh connections contains us, so we need to add a check against participant == me here

Reply:
Interesting, so in the pool we also maintain a NodeConnection to self? Shouldn't we filter that when we spawn the tasks?

Reply:
so right now, Mesh doesn't have a notion of `me: Participant`, so unable to filter out that connection right now. We could try adding it in via the `NodeStateWatcher` but likely won't behave well with initial KeyGen so I avoided adding it for now. Sync at least has the `me` info, so we can filter it there

Reply:
Okay, I think I have this covered now with a check in the sync task, right inside the spawned task of broadcast_sync. It's a bit ugly but in turn we keep to things simple in this PR, I guess.